app/policies/template_policy.rb

```diff
@@ -1,7 +1,10 @@
 class TemplatePolicy < ApplicationPolicy
   class Scope < ApplicationPolicy::Scope
     def resolve
-      permission_check('read', scope) ? scope.all : (raise Exceptions::NotAuthorizedError, "Read access not authorized for #{scope}")
+      return graphql_id_query if graphql_query_by_id?
```

Comment:
This line repeats in every scope's `resolve` method. We can move it in to its base implementation. In base's `resolve` method we can call `resolve_graphql_scope` and `resolve_native_scope`. And you only need to override `resolve_native_scope` for request and action policies.

Reply:
Actually you even don't need `resolve_graphql_scope`. The base scope's resolve method can be
```
  def resolve
    return graphql_id_query if graphql_query_by_id?
   resolve_collection_scope
  end

  def resolve_collection_scope
     raise Exceptions::NotAuthorizedError, "Read access not authorized for #{scope}" unless permission_check('read', scope)

    graphql_query? ? graphql_collection_query : scope.all
  end
```
Seems `resolve_collection_scope` is a better name.